specfile/specfile.py

```diff
@@ -111,6 +149,43 @@ def __exit__(
     ) -> None:
         self.save()
 
+    def __deepcopy__(self, memodict: Dict[int, Any]):
+        """
+        Deepcopies the object, handling file-like attributes.
+        """
+        specfile = self.__class__.__new__(self.__class__)
+        memodict[id(self)] = specfile
+
+        for k, v in self.__dict__.items():
+            if k == "_file":
+                continue
+            setattr(specfile, k, copy.deepcopy(v, memodict))
+
+        try:
+            path = Path(self._file.name)
+        except AttributeError:
+            # not a named file, try `getvalue()`
+            try:
+                specfile._file = type(self._file)(self._file.getvalue())
+            except AttributeError:
+                # last resort, try to instantiate without arguments
+                # and copy data
+                specfile._file = type(self._file)()
+                specfile._file.write(self._file.read())
+                specfile._file.seek(0)
+        else:
+            try:
+                specfile._file = path.open(
+                    mode=self._file.mode,
+                    encoding=self._file.encoding,
+                    errors=self._file.errors,
+                )
+            except AttributeError:
+                # files open in binary mode have no `encoding`/`errors`
+                specfile._file = path.open(mode=self._file.mode)
```

Comment:
Here we are treating `self._file` as named file again.

```suggestion
                specfile._file = path.open(mode=cast(FileIO, self._file).mode)
```